chemprop/utils/mixins.py

```diff
@@ -1,20 +1,57 @@
+"""Mixin classes for the project."""
+
 import inspect
 from typing import Any, Collection
 
 
 class ReprMixin:
     def __repr__(self) -> str:
-        items = self.get_params()
+        """Return a string representation of the object.
+
+        Returns
+        -------
+        str
+            The string representation of the object.
+        """
+        param_repr_list = []
+        for param_name, param_value in self.get_params(deep=False).items():
+            param_default = self._get_default_params()[param_name]
+            if param_value != param_default:
+                param_repr_list.append(f"{param_name}={repr(param_value)}")
+
+        non_default_params = ", ".join(param_repr_list)
+        return f"{self.__class__.__name__}({non_default_params})"
+
+    def _get_default_params(self) -> dict[str, Any]:
+        """Get the default parameters for initializing the class."""
+        sig = inspect.signature(self.__class__)
+        return {k: v.default for k, v in sig.parameters.items()}
+
+    def _get_parm_names(self) -> list[str]:
+        """Get the default parameters for initializing the class."""
+        return sorted(self._get_default_params().keys())
 
-        if len(items) > 0:
-            keys, values = zip(*items)
-            sig = inspect.signature(self.__class__)
-            defaults = [sig.parameters[k].default for k in keys]
-            items = [(k, v) for k, v, d in zip(keys, values, defaults) if v != d]
+    def get_params(self, deep: bool = True) -> dict[str, Any]:
+        """Get the current parameters of the class, required to initialize the class.
 
-        argspec = ", ".join(f"{k}={repr(v)}" for k, v in items)
+        This method mimics the behavior of `sklearn.base.BaseEstimator.get_params`.
 
-        return f"{self.__class__.__name__}({argspec})"
+        Parameters
+        ----------
+        deep : bool, default=True
+            If True, will return the parameters for this estimator and
+            contained subobjects that are estimators.
 
-    def get_params(self) -> Collection[tuple[str, Any]]:
-        return self.__dict__.items()
+        Returns
+        -------
+        dict[str, Any]
+            Parameter names mapped to their values.
+        """
+        out: dict[str, Any] = {}
+        for param_name in self._get_parm_names():
+            value = getattr(self, param_name)
+            if deep and hasattr(value, "get_params") and not isinstance(value, type):
+                deep_items = value.get_params().items()
+                out.update((param_name + "__" + k, val) for k, val in deep_items)
+            out[param_name] = value
+        return out
```

Comment:
```suggestion
        name_to_value = {}
        for name in self._get_default_params.keys():
            value = getattr(self, param_name)
            name_to_value[name] = value
            if deep and hasattr(value, "get_params") and not isinstance(value, type):
                _params = value.get_params()
                name_to_value.update((f"{name}__{k}", v) for k, v in _params.items())
            
        return name_to_value
```

Reply:
Looks like this function was copied straight from scikit-learn. Would probably be good to edit it to fit the style of Chemprop more as you have done. Could also be good to leave it identical to scikit-learn if that has worked for scikit-learn.